### PR TITLE
expression: rewrite builtin function: FORMAT

### PR DESCRIPTION
**Pull request by zz-jason**
to #4080

### CODE REVIEW

File: expression/integration_test.go [hunk 1]
```diff
@@ -834,6 +834,12 @@ func (s *testIntegrationSuite) TestStringBuiltin(c *C) {
 	result.Check(testkit.Rows("'aaaa' '' '\"\"' '\n\n'"))
 	result = tk.MustQuery(`select quote(0121), quote(0000), quote("中文"), quote(NULL);`)
 	result.Check(testkit.Rows("'121' '0' '中文' <nil>"))
+
+	// for format
+	result = tk.MustQuery(`select format(12332.1, 4), format(12332.2, 0), format(12332.2, 2,'en_US');`)
```
Comment (XuHuaiyu) [inline expression/integration_test.go]: add test cases for other locale here.

File: expression/builtin_string.go [hunk 2]
```diff
@@ -2450,51 +2451,71 @@ func (c *formatFunctionClass) getFunction(ctx context.Context, args []Expression
...
+	formatString, err := mysql.GetLocaleFormatFunction(locale)(x, d)
+	return formatString, err != nil, errors.Trace(err)
+}
+
+type builtinFormatSig struct {
+	baseStringBuiltinFunc
+}
+
+// evalString evals FORMAT(X,D).
+// See https://dev.mysql.com/doc/refman/5.6/en/string-functions.html#function_format
```
Comment (jackysp) [inline expression/builtin_string.go]: Maybe we should change the doc reference to version 5.7 here or in another pr as we claimed.

**Comment by XuHuaiyu** [comment]
LGTM

**Comment by zz-jason** [comment]
@jackysp PTAL

**Pull request merged**